CheckoutV2: Support processing channel and marketplace sub entity ID

AMHOL opened the pull request:
# Test Summary

## Unit
5034 tests, 74937 assertions, 0 failures, 0 errors, 0 pendings, 0 omissions, 0 notifications
100% passed

## Rubocop
728 files inspected, no offenses detected

## Remote
`bundle exec rake test:remote TEST=test/remote/gateways/remote_checkout_v2_test.rb`
43 tests, 110 assertions, 0 failures, 0 errors, 0 pendings, 0 omissions, 0 notifications
100% passed

therufs commented:
Hey @AMHOL, 

This looks great so far! Could we, just for completeness, go ahead and pop the updated options into the corresponding remote test (`test_successful_purchase_with_additional_options`) too?

AMHOL commented:
Thanks @therufs, I've amended the commit and force pushed to keep all changes in a single commit, see https://github.com/activemerchant/active_merchant/pull/4236/files#diff-997caba8ce528eaa5690041791c797b18f9727d588b5f26bee535dcf013f9469

jessiagee commented on lib/active_merchant/billing/gateways/checkout_v2.rb:
```diff
@@ -186,6 +188,17 @@ def add_3ds(post, options)
+      def add_marketplace_data(post, options)
+        if options[:marketplace]
+          post[:marketplace] = {}
+          post[:marketplace][:sub_entity_id] = options[:marketplace][:sub_entity_id] if options[:marketplace][:sub_entity_id]
```
Let's check for nil with `options[:marketplace]` before attempting to dig into it

AMHOL replied:
@jessiagee that is done on [line 196](https://github.com/activemerchant/active_merchant/pull/4236/files#diff-ac1e1ae60a6cc04111367e448e705b6e74104fd908e64d1ffe6c7237b5deb65dR196).

AMHOL replied:
All good @jessiagee, thanks for the review.

jessiagee reviewed (changes requested):
Hi AMHOL! Let's fix the one area I pointed out than rebase and squash commits and I'll merge you into AM. Thank you!

ajawadmirza reviewed (approved):
All tests passing. Good to go 👍

AMHOL commented:
TY @jessiagee and @ajawadmirza

jessiagee commented:
@AMHOL can you rebase this PR and squash the commits? Once that's done I'll approve and merge in. Thank you!

AMHOL commented:
@jessiagee I've rebased and updated the description with the new local test run and rubocop data, but remote tests are failing on `test_partial_refund(RemoteCheckoutV2Test)` with a `403` error, I think it may be something to do with the secret key I'm using as I get the same failure on the master branch too.

AMHOL commented:
@jessiagee remote tests are passing now:

43 tests, 110 assertions, 0 failures, 0 errors, 0 pendings, 0 omissions, 0 notifications
100% passed
